test/integration/targets/systemd/tasks/main.yml

```diff
@@ -72,4 +72,26 @@
               - result is failed
               - result is search("Could not find the requested service {{ fake_service }}")
 
+    - name: get a list of inactive services
+      shell: systemctl --all | fgrep 'inactive' | awk '{print $1}' | sed 's/\.service//g' | fgrep -v '.' | egrep ^[a-z]
+      register: inactive_names
+    - debug: var=inactive_names
+
+    - name: try-restart result should restart serivce when it is running
+      systemd:
+          name: "{{ running_names.stdout_lines|random }}"
+          state: try-restarted
+      register: running_try_restart
+
+    - name: try-restart result should not restart service when it is not running
+      systemd:
+          name: "{{ inactive_names.stdout_lines|random }}"
+          state: try-restarted
+      register: inactive_try_restart
+
+    - assert:
+          that:
+              - 'running_try_restart.changed'
+              - 'not inactive_try_restart.changed'
```

Comment:
```suggestion
              - 'running_try_restart is changed'
              - 'inactive_try_restart is not changed'
```

Reply:
Fixed in https://github.com/ansible/ansible/pull/71172/commits/a7693bc58abd283ea903b345c18c1e743c7012c3.